### ISSUE TITLE
Add a proper export/import system using the android storage access framework for saves

### ISSUE THREAD

**Issue by Vixeliz** [skyline-emu/skyline#1341]
I think this would be beneficial and future proof as getting access to android/data is not actually supported and on some phones such as my own I can't even copy files out of there while using one of the file managers that let's you in the directory. I'm aware you can access from pc however i think this is inconvenient as saves should be backed up frequently.

**Comment by PixelyIon** [comment]
This was the original plan and exposing the saves in the external storage directory was simply a temporary stopgap.

**Comment by Jhonanftr** [comment]
Bro tengo un Xiaomi redmi note 9 y no me corren los juegos ya tiene instalada las keys e intentado de todo y no me agarra

**Comment by DanXG790** [comment]
> Bro tengo un Xiaomi redmi note 9 y no me corren los juegos ya tiene instalada las keys e intentado de todo y no me agarra

Mali GPU doesn't work for the moment, Give them time and in the future they will be supported

**Comment by Vixeliz** [comment]
Is there an eta on this? I have no ways to get my save from the directory I've tried adb, mtp, and on device none work. All say permission deny or just fail. I've spent around 9 hours on celeste ik it's my own fault since the emulator is expirmental but any ideas?

**Comment by PixelyIon** [comment]
Sorry but at the moment we have no ETA for this feature and are busy with a lot of other stuff, it's really weird that you can't access the external files directory even with ADB/MTP. I assume you've tried both `adb pull` and `adb shell` (and tried `cd`-ing into the directory) but if neither of them work there's very little which can be done.

**Comment by Vixeliz** [comment]
So originally I was trying to use adb pull. However it appears the adb run-as command can cat the files to get the contents which can be piped into something else

**Comment by PixelyIon** [comment]
Yep, you can use those to get it out but getting things back in is a bit tougher iirc.

**Comment by Vixeliz** [comment]
> Yep, you can use those to get it out but getting things back in is a bit tougher iirc.

Adb push worked fine for me. Still annoying though looking forward to document provider

**Comment by Vixeliz** [comment]
I would like to update anyone looking at this adb push appears to look like it works at first but seems to prevent skyline from writing to the save file. So can still use cat to get saves backed up but you are correct importing doesnt work on devices with read and write problems for android/data

**Comment by Vixeliz** [comment]
Would be fixed by #1402
